src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -394,6 +394,9 @@ export class EditDataGridPanel extends GridParentComponent {
 		undefinedDataSet.dataRows = undefined;
 		undefinedDataSet.resized = new Emitter();
 		self.placeHolderDataSets.push(undefinedDataSet);
+		if (self.placeHolderDataSets[0]) {
```

Comment:
This is always going to be true since the line above has us pushing the new undefinedDataSet into the array. So at the very least this is unnecessary.